### DISCUSSION TITLE
Defining ramps in protocols

### DISCUSSION THREAD

**Discussion by FarmHJ** [MichaelClerx/myokit#1081] [Q&A]
Hi Michael,

I have a problem with defining a protocol with ramp. I tried using the method discussed in https://github.com/myokit/myokit/issues/802, but am still getting myokit SimulationError. I tried pacing the protocol at -80 mV and -120 mV during the ramp step that starts at -120 mV and ends at -80 mV. Both did not work. A screenshot of the error message:
![myokit_sim_err](https://github.com/user-attachments/assets/bed1e8f3-ea1a-4f05-bf15-b010f6291818)

**Comment by MichaelClerx** [comment]
Hi @FarmHJ ,
For some reason CVODE is having trouble calculating the solution to the requested tolerance. Are the rates unusually fast when the failure occurs? Or is there something else happening, numerically, that's making it hard to make the next time step?

**Comment by MichaelClerx** [comment]
Maybe calculate all rates by hand, and see if any are incredibly fast. If so, that points towards a way you can improve your prior.

If you want a really quick-and-dirty solution, you could say that if CVODE can't solve it, then the given parameters are unlikely to be right. But that's (A) a bit dangerous because you lose control / awareness of what gets rejected and why; and (B) not very efficient
